### ISSUE TITLE
Default Editor on Windows

### ISSUE THREAD

**Issue by MinchinWeb** [bram85/topydo#120]
I've been trying to decide what should be done for the editor on Windows. Using the Environment variables seems an odd way to go on a Windows machine, and the default of `vi` isn't found by default on Windows systems. There seem to be two default text editors on Windows: `notepad`, which doesn't recognize the line endings used by `topydo`, and Powershell ISE, which would work, but is only available on newer versions (Windows 10 and up?) and is more designed for building powershell scripts than file editing.

What I would be inclined to do is add another option (editor) to `topydo` configuration file, and refer here before looking to the environmental variable or trying `vi`. Thoughts?

**Comment by bram85** [comment]
Yes, I had the similar idea to have this in the configuration file but it never materialized.

**Comment by wenindoubt** [comment]
This is kind of, but not really related to this specific issue, but I figure I'll post it here instead of opening a new issue related to editors.

As we know, the default $EDITOR is vi. I am a heavy vim user, so for those of you who want to change the $EDITOR in *nix systems, I type the following:

`EDITOR=vim topydo edit <NUMBER>`

An easier way is to alias it in your ~/.{bashrc,zshrc} file.

`alias t="EDITOR=vim topydo"`

**Comment by bram85** [comment]
Or `export EDITOR=vim`, so all applications will benefit.

**Comment by bram85** [comment]
The `master` branch allows you to configure the editor in several ways. See
the message in commit ebb3b4c.

Do you think this issue is sufficiently addressed for Windows?

**Comment by MinchinWeb** [comment]
At a glance, it looks good. That said, I haven't yet had the chance to test it yet.

**Comment by bram85** [comment]
I'll close the issue for now, please reopen if the issue has not yet been addressed.

**Comment by MinchinWeb** [comment]
I was able to test it (I used the current master branch). The short answer is edit itself work as expected. I would suggest updating the documentation with the following suggested editor for Windows:

```
[edit]
editor = "C:\Program Files (x86)\Notepad++\notepad++.exe" -multiInst -nosession
```

This requires the use to install Notepad++, but deals with Linux text files much better than Windows' built-in Notepad.

A couple of oddities I noticed:

- using Windows' build-in text editor (Notepad) doesn't render newlines. This is typical when dealing with Linux-formatted files, but does cause some weird issues (e.g. if you edit one item and append to the end of the line, anything appended with be added as a new item because it is after an invisible newline). If this was solved, maybe Notepad should be the default editor on Windows systems? (as `vim` does not come with Windows systems)
- an item edited is removed from the list and then re-added to the bottom of the list. This has the side-effect of changing item numbers, and probably means that setting items as dependent on one another during editing with break
- even with `auto_creation_date = 1`, creation dates are not added to additional items added during editing. I can't decide whether this should be considered a bug or expected behaviour. 
- edited items are not "processed" for dependent tags (like "before:8"). I think this should be done, but with items numbers changing, this will be hard to do in a user-friendly manner

**Comment by bram85** [comment]
Reopening so your feedback won't get out of sight. I need to investigate.

**Comment by davesteele** [comment]
Todo - update the TiddlyWiki to document the editor configuration parameter, per https://github.com/topydo/topydo/issues/120#issuecomment-280091373.

**Comment by MinchinWeb** [comment]
In the intervening years, I think Windows Notepad has been updated to deal with Linux-style line endings properly. That's probably Windows 10 only.
